File: #29 jokenpo/jokenpo_test.rb
require "minitest/autorun"
require_relative "2018-1/elisson ruby/opcoes"
require_relative "2018-1/elisson ruby/jokenpo"

class JokenpoTest < Minitest::Test
  def setup
    @tesoura = Tesoura.new
    @papel   = Papel.new
    @pedra   = Pedra.new
  end

  def test_jogador_1_vence
    assert_equal "Jogador 1 Venceu. Papel Vence Pedra",
                  Jokenpo.new(@papel, @pedra).vencedor
    assert_equal "Jogador 1 Venceu. Pedra Vence Tesoura",
                  Jokenpo.new(@pedra, @tesoura).vencedor
    assert_equal "Jogador 1 Venceu. Tesoura Vence Papel",
                  Jokenpo.new(@tesoura, @papel).vencedor
  end

  def test_jogador_2_vence
    assert_equal "Jogador 2 Venceu. Papel Vence Pedra",
                  Jokenpo.new(@pedra, @papel).vencedor
    assert_equal "Jogador 2 Venceu. Pedra Vence Tesoura",
                  Jokenpo.new(@tesoura, @pedra).vencedor
    assert_equal "Jogador 2 Venceu. Tesoura Vence Papel",
                  Jokenpo.new(@papel, @tesoura).vencedor
  end

  def test_empate
    assert_equal "Empate",
                  Jokenpo.new(@pedra, @pedra).vencedor
    assert_equal "Empate",
                  Jokenpo.new(@papel, @papel).vencedor
    assert_equal "Empate",
                  Jokenpo.new(@tesoura, @tesoura).vencedor
  end
end

File: #29 jokenpo/JokenpoTest.java
import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

class JokenpoTest {
	Pedra 	pedra;
	Papel 	papel;
	Tesoura tesoura;

	@BeforeEach
	void setUp() throws Exception {
		pedra 	= new Pedra();
		papel 	= new Papel();
		tesoura = new Tesoura();
	}

	@Test
	void testPapelVencePedra() {
		assertTrue(papel.vence(pedra));
	}
	@Test
	void testPapelNaoVenceTesoura() {
		assertFalse(papel.vence(tesoura));
	}
	@Test
	void testPedraVenceTesoura() {
		assertTrue(pedra.vence(tesoura));
	}
	@Test
	void testPedraNaoVencePapel() {
		assertFalse(pedra.vence(papel));
	}
	@Test
	void testTesouraVencePapel() {
		assertTrue(tesoura.vence(papel));
	}
	@Test
	void testTesouraNaoVencePedra() {
		assertFalse(tesoura.vence(pedra));
	}
	@Test
	void testJokenpoRetornaVencedor() {
		assertEquals("Jogador 1 Venceu", new Jokenpo(pedra, tesoura).vencedor());
		assertEquals("Jogador 1 Venceu", new Jokenpo(tesoura, papel).vencedor());
		assertEquals("Jogador 1 Venceu", new Jokenpo(papel, pedra).vencedor());
		assertEquals("Jogador 2 Venceu", new Jokenpo(tesoura, pedra).vencedor());
		assertEquals("Jogador 2 Venceu", new Jokenpo(papel, tesoura).vencedor());
		assertEquals("Jogador 2 Venceu", new Jokenpo(pedra, papel).vencedor());
	}
	@Test
	void testJokenpoRetornaEmpateQuandoOpcoesSaoIguais() {
		assertEquals("Empate", new Jokenpo(tesoura, tesoura).vencedor());
		assertEquals("Empate", new Jokenpo(papel, papel).vencedor());
		assertEquals("Empate", new Jokenpo(pedra, pedra).vencedor());
	}
}
